src/SubPoolFactory.sol

```diff
@@ -144,9 +154,42 @@ contract SubPoolFactory is Auth, ISubPoolFactory {
         subPoolData[pool] = subpoolData;
     }
 
+    /// @notice Update solver membership.
+    function updateSolverMembership(address solver, bool add) external {
+        address pool = msg.sender;
+        SubPoolData memory subpoolData = subPoolData[pool];
+        if (subpoolData.collateral == address(0)) revert SubPoolFactory__UnknownPool();
+
+        if (add) {
+            // check if solver already has a pool of its own
+            if (subPoolData[poolOf(solver)].collateral != address(0)) {
+                revert SubPoolFactory__SolverHasActiveMembership();
+            }
+            // check if solver has a membership of some other pool
+            if (solverBelongsTo[solver] != address(0)) revert SubPoolFactory__SolverHasActiveMembership();
+            solverBelongsTo[solver] = pool;
+            emit UpdateSolverMembership(pool, solver, true);
+        } else {
+            // check that the solver is member of the given pool
+            if (solverBelongsTo[solver] != pool) revert SubPoolFactory__SolverNotAMember();
+            solverBelongsTo[solver] = address(0);
+            emit UpdateSolverMembership(pool, solver, false);
```

Comment:
Should a solver be allowed to leave its own pool?

Reply:
A solver wont be a member of its own pool, i.e. `solverBelongsTo` wont be pointing to solver's pool. So they can't really leave it. Ownership is permanent because the pool address is tied/derived from owner's address.

Reply:
I see, makes sense. Then, I'd try to separate `solver` as pool creators and `solver` as addresses that belong to a pool with a different name. Maybe the former could be `underwriter`, or `guarantor`?

Also, am I right in assuming then that the following line from `create` is technically unnecessary?
```
if (solverBelongsTo[msg.sender] != address(0)) revert SubPoolFactory__SolverHasActiveMembership()
```

Reply:
It is necessary, it is preventing solvers that are already part of some pool from spawning their own pool

Reply:
Is it an issue if an address that is the underwriter of a pool is also a solver for another pool? I don't expect this to happen in practice, but even if it happened I don't see what could go wrong. Am I missing something?

Reply:
no, but just trying to have consistent state. i.e. if that map is set we can be sure the solver is not an underwriter themselves.

Reply:
With the changes in the code that allowed multiple solvers to join a single pool, I don't think it makes sense to see the pool creator as a solver anymore, the pool creator has just become a manager. With this is mind, preventing a solver from being an underwriter/pool creator/manager doesn't seem much more helpful than preventing a solver to become an owner (as in `isOwner[owner] == true`). That is, I don't see any need for "consistency" between solvers and underwriters. Unless there is some concrete link between the two that I'm missing?

Reply:
removed the check, but i made the pool creator its own solver on pool creation, seems like a good default to have